tensorboard/components/tf_ng_tensorboard/header/containers/header.component.test.ts

```diff
@@ -145,4 +150,24 @@ describe('header.component', () => {
     expect(dispatch).toHaveBeenCalledTimes(1);
     expect(dispatch).toHaveBeenCalledWith(changePlugin({plugin: 'bar'}));
   });
+
+  it('does not keep observer after getting the most recent value', async () => {
+    const dispatch = spyOn(store, 'dispatch');
+    const fixture = TestBed.createComponent(HeaderComponent);
+    fixture.detectChanges();
+
+    const [, debugEl] = fixture.debugElement.queryAll(By.css('.mat-tab-label'));
+    debugEl.nativeElement.click();
+    fixture.detectChanges();
+    await fixture.whenStable();
+    expect(dispatch).toHaveBeenCalledTimes(1);
+
+    // If an observer is still listening, this should cause the store to dispatch.
```

Comment:
Sorry, I don't follow.  Which observer were you worried might still be listening, but in fact should not be?  Also, `setActivePlugin()` calls `setState()` but not `dispatch()`, so I'm not sure this is actually testing what you meant.

Reply:
Oh, is this about the `take(1)` at header.component.ts:70?  If so refactoring it per my comment there may obviate the need for the test.

Reply:
looks like the test was obsoleted in the refactoring.